test/jdk/java/lang/Math/RoundTests.java

```diff
@@ -29,16 +29,17 @@
 public class RoundTests {
     public static void main(String... args) {
         int failures = 0;
-
-        failures += testNearFloatHalfCases();
-        failures += testNearDoubleHalfCases();
-        failures += testUnityULPCases();
-        failures += testSpecialCases();
-
-        if (failures > 0) {
-            System.err.println("Testing {Math, StrictMath}.round incurred "
-                               + failures + " failures.");
-            throw new RuntimeException();
+        for (int i = 0; i < 100000; i++) {
```

Comment:
Is there an idiom to trigger the auto-vectorization, perhaps using command line arguments, that doesn't bloat the running time of this test?

Reply:
IMO RoundTests should have a explicit @run tag without any VM options as well.

Do the added VM options run on all platforms in question? What is the approximate time to run the test run compared to before?

Reply:
Hi @jddarcy , 

Test has been modified on the same lines using generic options which manipulate compilation thresholds and agnostic to target platforms.

 * @run main/othervm -XX:Tier3CompileThreshold=100 -XX:CompileThresholdScaling=0.01 -XX:+TieredCompilation RoundTests

Verified that  RoundTests::test* methods gets compiled by c2.
Test execution time with and without change is almost same ~7.80sec over Skylake-server.

Regards

Reply:
To be more explicit, the existing RoundTests.java test runs in a fraction of a second.  The updated test runs many times slower, even if now under 10 second, at least on some platforms.

Can something closer to the original performance be restored?

As a tier 1 library test, these tests are run quite frequently.

Reply:
Hi @jddarcy , 
Earlier none of the test methods in  RoundTests.java were compiled on account of low invocation count,  a loop with 2000 iterations under the influence controlled compilation threshold now triggers tier4 compilation of test points. I did several runs in Skylake machine with patch and without patch and could see no perceptible difference in runtime due to modification.

I have further reduced the invocation count and compile threshold.

Thanks